Resetting default values

ppxma7 opened the pull request:
Hi Julien
Thanks for catching those lines I missed - I was sure I fixed them, but something must have gone wrong when I pushed. I closed the previous pull request and started a new one afresh. Hopefully this time everything will merge. Let me know if not
Best
Michael

julienbesle commented:
Ok. Great. 
You can easily see all the changes that result from applying all commits in the pull request by going to the "Files changes" tab above and looking at the changes resulting from all commits (or from a subset of these commit). This is how I noticed that you had missed some lines.
I think maybe for future pull requests, try to include fewer commits. And also, it would be helpful to have more explanation about what each commit achieves, especially if it affects the general mrTools code. Obviously it's less crucial for your own pRF_somato plugin (although still useful if you have to go back yourself to check for bugs)

I still have some questions about the remaining changes before I pull the request. I will comment on each individual change as previously.

Cheers

julienbesle commented on mrLoadRet/Plugin/GLM_v2/newGlmAnalysis/hrfDoubleGamma.m:
```diff
@@ -16,6 +16,12 @@
   sampleDelay=sampleDuration/2;
 end
 
+if sampleDuration == 1000
```
It seems that this issue comes from having stim times in milliseconds instead of seconds, and it looks like this fix is only going to work in a very specific case where the duration has been set to 1000 ms (instead of 1s). This fix wouldn't work if it has been set to 2000ms.
I think the question here is why is the sample duration field was set to 1000 to start with? So I suspect that the fix should be somewhere else and there isn't really a bug here.
This change is unlikely to affect anything for other users because no one will ever actually use a duration of 1000s, but it's not very general.

julienbesle commented on mrLoadRet/Plugin/GLM_v2/newGlmAnalysis/glmPlot_edit.m:
```diff
@@ -0,0 +1,852 @@
+function glmPlot_edit(thisView,overlayNum,scanNum,x,y,s,roi)
```
what is the difference between glmPlot_edit and the original glmPlot? Are there some new features that could benefit people who use glmPlot, and can these features be integrated into glmPlot instead?
Otherwise, if it was just for convenience to do soemthing quickly I'd rather not add this function to the main mrTools repository. 
Instead you could set the "interrogatorPaths" preferrence to add an alternative folder where you can add your own interrogator function (using mrSetPref)

julienbesle commented:
Another thing I'm wondering is why the following mexmaci64 executables have been modified:
mrUtilities/ImageProcessing/corrDn.mexmaci64
mrUtilities/ImageProcessing/myCinterp3.mexmaci64
mrUtilities/ImageProcessing/upConv.mexmaci64
mrUtilities/mrFlatMesh/assignToNearest.mexmaci64
mrUtilities/mrFlatMesh/dijkstra.mexmaci64

Did you have to recompile them? 
I'm a bit worried that if these files might not work on someone else's 64bit Mac. They are not supposed to be machine-specific but they are apparently specific  to certain Matlab versions
I would be a good idea to test whether the old versions of the files (the ones before you did the change) work for you to see if they really need to change.

ppxma7 commented:
Hi Julien
I have removed glmPlot_edit. There was a display problem a while back, which I never had time to fix, so had to comment some parts of the original code and so made this altered form. I have removed it from this pull request. 

I have hopefully made the sampleDUration bug clearer now in hrfDoubleGamma

Re: the mex files, Yes, I had to recompile them on my Mac using Mojave. None of the original mex files worked on my machine on MATLAB post 2016b. I wrote a script in the Nottingham repo to recompile these as I had to do it so often for others in the centre who updated their Macs and wanted to use mrTools. Perhaps I can keep these compiled versions separate from this pull request and keep them elsewhere?

julienbesle commented:
Ok. The millisecond/second conversion is more general now. It's still a question why durations would be specified in milliseconds... but it's ok to include the fix because it's unlikely to impact anything else since duration will never be as large as 1000s
For the mex files, I think it would be better not to include them in case someone uses an older version of MacOS or Matlab. 
But it would be useful to catch errors when trying to run an outdated mex file and re-compilation is needed. The message could include instructions on how to compile the mex file. The only issue with this is that the test would have to be included in all calls to any mex function in mrTools... A slighlty better solution would be to write a wrapper function that tries to run the mex function passed as an argument, issues a warning if it catches an error and runs the corresponding m file instead. Maybe a project for later.
In any case, once you've removed the updated mex files from the pull request (i.e. replaced them with the original versions), I can merge it.

julienbesle commented:
I just saw that there is an mlrMake.m function that seems to recompile all necessary mrTools mex files (only on a Mac), in case it's useful...

ppxma7 commented:
Hi Julien, trying to reset the mex files to the default ones you have on your repo. I deleted my recompiled ones and copied over yours, but, the pull request still says that they've been modified - any ideas?

julienbesle commented:
What does it say when you try "git diff" on one of the mex files?

julienbesle commented:
Actually, I just checked and the mex files are do not show up anymore in the "changed files" tab, so I can merge the pull request
